web/views.py

```
@@ -5101,6 +5102,168 @@ def toggle_course_status(request, slug):
+            achievement.icon = icon_url
+            achievement.save()
+
+            # Pass achievement ID, wallet address, and icon URL to NFT minting service
+            nft_badge = mint_nft_badge(achievement_id, wallet_address)
+
+            if nft_badge:
+                messages.success(request, f"NFT badge successfully minted and sent to {wallet_address}!")
+                return redirect("achievement_detail", achievement_id=achievement_id)
+            else:
+                messages.error(request, "Failed to mint NFT badge. Please try again later.")
+        except Exception as e:
```

Ruff (0.8.2)

Do not catch blind exception: `Exception`

(BLE001)